packages/postgres-database/tests/test_clusters.py

```diff
@@ -0,0 +1,129 @@
+# pylint: disable=no-value-for-parameter
+# pylint: disable=redefined-outer-name
+
+from typing import Awaitable, Callable
+
+import psycopg2
+import pytest
+import sqlalchemy as sa
+from aiopg.sa.engine import Engine
+from aiopg.sa.result import ResultProxy
+from pytest_simcore.helpers.rawdata_fakers import random_user
+from simcore_postgres_database.models.cluster_to_groups import cluster_to_groups
+from simcore_postgres_database.models.clusters import ClusterType, clusters
+from simcore_postgres_database.models.users import users
+
+
+@pytest.fixture(scope="function")
+async def user_id(pg_engine: Engine) -> int:
+    async with pg_engine.acquire() as conn:
+        # a 'me' user
+        uid = await conn.scalar(
+            users.insert().values(**(random_user())).returning(users.c.id)
+        )
+    yield uid
+    # cleanup
+    async with pg_engine.acquire() as conn:
+        # a 'me' user
+        uid = await conn.execute(users.delete().where(users.c.id == uid))
+
+
+@pytest.fixture
+async def user_group_id(pg_engine: Engine, user_id: int) -> int:
+    async with pg_engine.acquire() as conn:
+        primary_gid = await conn.scalar(
+            sa.select([users.c.primary_gid]).where(users.c.id == user_id)
+        )
+    return primary_gid
+
+
+@pytest.fixture
+async def create_cluster(pg_engine: Engine) -> Callable[..., Awaitable[int]]:
+    cluster_ids = []
+
+    async def creator(**overrides) -> Awaitable[int]:
+        insert_values = {
+            "name": "default cluster name",
+            "type": ClusterType.ON_PREMISE,
+            "description": None,
+        }
+        insert_values.update(overrides)
+        async with pg_engine.acquire() as conn:
+            cluster_id = await conn.scalar(
+                clusters.insert().values(**insert_values).returning(clusters.c.id)
+            )
+        cluster_ids.append(cluster_id)
+        return cluster_id
+
+    yield creator
+
+    # cleanup
+    async with pg_engine.acquire() as conn:
+        await conn.execute(clusters.delete().where(clusters.c.id.in_(cluster_ids)))
+
+
+async def test_cluster_without_owner_forbidden(
+    create_cluster: Callable[..., Awaitable[int]]
+):
+    with pytest.raises(psycopg2.errors.NotNullViolation):
```

Comment:
NOTE: see ``simcore_postgres_database.errors``. psycopg2 errors are not explicit and sometimes difficult to find. IN that module we try to collect some of the errors that we are using. I propose we use this as a indirection for all database related errors

Reply:
so if I get it right, I should indirect all the errors through ```simcore_postgres_database.errors```? Done.